Update system test docs to include new Elastic Agents (#1830)
elastic/elastic-package: docs/howto/system_testing.md

To define a system test we must define configuration on at least one level: a package or a data stream's one.

> If the package does not require a service for testing, there is no need to define any service deployer.

Author: Note about not requiring anymore a service deployer.

If the package requires a service, we must define the configuration for deploying a package's integration service.

> We can define it on either the package level:

Author: I prefer your suggestion, I'll update it, thanks!

| Option | Type | Required | Description |
|---|---|---|---|
| agent.linux_capabilities | array string | | Linux Capabilities that must be enabled in the system to run the Elastic Agent process. |

> | agent.pid_mode | string | | Turns on sharing between container and the host operating system the PID address space. |

Reviewer: Maybe we should mention the possible values here. And phrase it in a way that is less bound to containers.
```suggestion
| agent.pid_mode | string | | Controls access to PID namespaces. When set to `host`, the agent will have access to the PID namespace of the host. |
```

| Option | Type | Required | Description |
|---|---|---|---|

> | agent.linux_capabilities | array string | | Linux Capabilities that must been enabled in the system to run the Elastic Agent process. |

Reviewer: ```suggestion
| agent.linux_capabilities | array string | | Linux Capabilities that must be enabled in the system to run the Elastic Agent process. |
```

| agent.pid_mode | string | | Turns on sharing between container and the host operating system the PID address space. |
| agent.ports | array string | | List of ports to be exposed to access to the Elastic Agent.|
| agent.runtime | string | | Runtime to run Elastic Agent process. |

> | agent.pre_start_script.language | string | | Programming language of the provisioning script. Default: `bash`.|

Reviewer: ```suggestion
| agent.pre_start_script.language | string | | Programming language of the pre-start script, executed before starting the agent. Default: `bash`.|
```

> | agent.pre_start_script.language | string | | Programming language of the provisioning script. Default: `bash`.|

Reviewer: Maybe we should also default it to `sh`, given that it is what is used in the provisioning script?

Author: I mixed the default languages for both scripts.
In the pre-start script, it must be `sh`. This is the only supported for now.
For the provisioning script, there is no restriction and the default one is bash.

I'll update those default values.

Reviewer: Could we make both to be `sh` by default? It looks weird to have different defaults for them.

Author: Yes, it could be done like that, no problem.
I'll update elastic-package and package-spec :+1:

Author: Indeed, in code it was set for both scripts `sh` as default language:
- https://github.com/elastic/elastic-package/blob/7b156a51b8db9f300d4ba2fb67d6bcea9b744b42/internal/testrunner/runners/system/test_config.go#L113
- https://github.com/elastic/elastic-package/blob/7b156a51b8db9f300d4ba2fb67d6bcea9b744b42/internal/agentdeployer/info.go#L16

> | agent.pre_start_script.contents | string | | Code to run as a provisioning script. |

Reviewer: ```suggestion
| agent.pre_start_script.contents | string | | Code to run before starting the agent. |
```

| agent.provisioning_script.language | string | | Programming language of the provisioning script. Currently, just supported `sh`. |

> | agent.provisioning_script.contents | string | | Code to run as a provisioning script. |

Reviewer: ```suggestion
| agent.provisioning_script.contents | string | | Code to run as a provisioning script to customize the system where the agent will be run. |
```

> | agent.user | string | | User that runs the Elastic Agent proces. |

Reviewer: ```suggestion
| agent.user | string | | User that runs the Elastic Agent process. |
```

| service_notify_signal | string |  | Signal name to send to 'service' when the test policy has been applied to the Agent. This can be used to trigger the service after the Agent is ready to receive data. |
| skip.link | URL |  | URL linking to an issue about why the test is skipped. |
| skip.reason | string |  | Reason to skip the test. If specified the test will not execute. |

> | skip_ignored_fields | array string |  | List of fields to be skipped when perfoing mapping validation. |

Author: Adding field introduced in #1738

Reviewer: Thanks!

> | skip_ignored_fields | array string |  | List of fields to be skipped when perfoing mapping validation. |

Reviewer: ```suggestion
| skip_ignored_fields | array string |  | List of fields to be skipped when performing validation of fields ignored during ingestion. |
```
